### DISCUSSION TITLE
Replacement for yq3 read

### DISCUSSION THREAD

**Discussion by chrischu** [mikefarah/yq#982] [Q&A]
I'm currently in the process of migrating to yq 4, but I'm struggling to convert my read calls. I'm using read like this: `yq.exe read (<path>) --printMode (p|pv)'. So essentially I want these features:

Given the input YAML:
```yaml
a:
  b:
    c: v1
    d: v2
```
* For values selected by <path> (or every value if no path) list their full path:
Previously used: `yq.exe read ** --printMode p`
```yaml
a.b.c
a.b.d
```

* For all values selected by <path> (or every value if no path)  list their full path & value:
Previously used: `yq.exe read ** --printMode pv`
```yaml
a.b.c: v1
a.b.d: v2
```

Is there an easy way to achieve the same with yq 4?

**Comment by mikefarah** [comment]
There are two ways:

You can use the props output:
```
yq e -o=props examples/data1.yaml
```

(though this doesn't quite match the format you've got);

and you can use the `path` and the `join` operator:

```
yq e '.. | select(tag != "!!map") | select(tag != "!!seq")  | {(path | join(".")): .}
```

- `..` matches all nodes, 
- `select(..)` we filter to select only nodes that aren't maps or arrays
- then we create a new object that has the path, joined by dots as the key, and the value `.` as the value
-
